### PR TITLE
Return error when schema is not available

### PR DESCRIPTION
**Pull request by soltysh**
**What type of PR is this?**
/kind bug
/priority important-longterm
/sig cli

**What this PR does / why we need it**:
If schema is nil or in general not available on server, just return error instead of panic-ing. 

**Which issue(s) this PR fixes**:
This was noticed in https://github.com/kubernetes/kubernetes/pull/71192

**Special notes for your reviewer**:
/assign @liggitt @sttts @roycaihw

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
add a test and fix bazel?

File: pkg/kubectl/explain/explain.go [hunk 1]
```diff
@@ -63,6 +64,9 @@ func PrintModelDescription(fieldsPath []string, w io.Writer, schema proto.Schema
 	if err != nil {
 		return err
 	}
+	if schema == nil {
```
Comment (apelisse) [inline pkg/kubectl/explain/explain.go]: the four-states-return-value-but-maybe-two? in go is quite terrible imo. I suspect it's a bug that we get a nil schema without an error?

Comment (apelisse) [reply pkg/kubectl/explain/explain.go]: what is the situation that triggered the panic?

Comment (soltysh) [reply pkg/kubectl/explain/explain.go]: I was considering that approach a viable fix, I'll look what's possible. This was found in https://github.com/kubernetes/kubernetes/pull/71192#discussion_r254784547 when we don't have an up-to-date information about CRD.

Comment (apelisse) [reply pkg/kubectl/explain/explain.go]: The problem is that this fix hides the actual problem. I'd rather not implement it this way. I had a discussion with @roycaihw and we'll try to improve the empty object instead.

Comment (liggitt) [reply pkg/kubectl/explain/explain.go]: I don't think it hides it… eero ring is better than panicking here, right?

Comment (apelisse) [reply pkg/kubectl/explain/explain.go]: Well, I might be wrong, but I still don't think it should happen. In which case I'm in general in favor of panics, because they tend to force people to do the right thing.

Comment (soltysh) [reply pkg/kubectl/explain/explain.go]: I prefer not to have panics, honestly ;)

Comment (liggitt) [reply pkg/kubectl/explain/explain.go]: > I prefer not to have panics, honestly ;)

agree

Comment (apelisse) [reply pkg/kubectl/explain/explain.go]: Let me rephrase. I don't want a panic, I just don't think we should have a nil schema in the first place.

**Comment by soltysh** [comment]
/hold
Alternative solution in https://github.com/kubernetes/kubernetes/pull/74029 @apelisse ptal

**Comment by soltysh** [comment]
The alternative was accepted.
/close
